### PR TITLE
[WFCORE-5526] Removal of legacy security realms.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.redhat.com/browse/WFCORE-5526

### CODE REVIEW

**Comment by bstansberry** [comment]
I believe the merge conflicts are just because the core 17.0.0.Beta5 release resulted in version bumps in 2 poms that this PR deletes.

File: server/src/main/java/org/jboss/as/server/mgmt/domain/HostControllerConnectionService.java [hunk 1]
```diff
@@ -81,7 +80,7 @@
 
     private final URI connectionURI;
     private final String serverName;
-    private final String userName;
+    private final String userName = null; // TODO This likely needs to be further visited.
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/mgmt/domain/HostControllerConnectionService.java:83]: JIRA?

Comment (darranl) [reply server/src/main/java/org/jboss/as/server/mgmt/domain/HostControllerConnectionService.java:83]: I believe this one is under this issue to follow up https://issues.redhat.com/browse/WFCORE-5544

**Review by jmesnil — APPROVED** [review]
🍾 !

**Comment by jmesnil** [comment]
@darranl thanks

**Pull request merged**